toolkit/tools/imagecustomizer/docs/clone-rpm-repo.md

```diff
@@ -0,0 +1,177 @@
+# Cloning an RPM repo
+
+By default, the image customizer tool uses the base image's inbuilt repo files for where
+to source RPMs from. The Azure Linux default repos typically point to
+packages.microsoft.com (PMC).
+
+PMC is regularly updated with bug fixes and feature updates for packages. So, if an
+image customization config includes package install or updates, then a run on one day
+may produce a different result than a run on another day since PMC might have been
+updated in between runs. This behavior may be perfectly fine (or even desirable) for
+some users. However, other users may require more stable builds that don't change based
+on state of an external resource (e.g. PMC). For such users it can be useful to make a
+clone of PMC.
+
+## Cloning a repo to a local directory
+
+1. Acquire the `dnf reposync` and `dnf download` commands.
+
+   Azure Linux 2.0 and 3.0:
+
+   ```bash
+   sudo tdnf -y install dnf-utils
+   ```
+
+   Ubuntu 24.04:
+
+   ```bash
+   sudo apt update -y
+   sudo apt install -y dnf-plugins-core
+   ```
+
+2. Select the repo URL:
+
+   | Azure Linux Version | Arch   | URL                                                               |
+   | ------------------- | ------ | ----------------------------------------------------------------- |
+   | 2.0                 | x86_64 | https://packages.microsoft.com/cbl-mariner/2.0/prod/base/x86_64/  |
+   | 2.0                 | ARM64  | https://packages.microsoft.com/cbl-mariner/2.0/prod/base/aarch64/ |
+   | 3.0                 | x86_64 | https://packages.microsoft.com/azurelinux/3.0/prod/base/x86_64/   |
+   | 3.0                 | ARM64  | https://packages.microsoft.com/azurelinux/3.0/prod/base/aarch64/  |
+
+   For example:
+
+   ```bash
+   REPO_URL="https://packages.microsoft.com/azurelinux/3.0/prod/base/x86_64/"
+   ```
+
+3. Clone PMC.
+
+   If you want clone all of PMC, then run:
+
+   ```bash
+   dnf reposync --repofrompath "azurelinux,$REPO_URL" --repo azurelinux --newest-only
```

Comment:
nit: maybe also make the repo local target folder be a variable and give it a date (i.e. `~/repos/azurelinux-0901`).

Reply:
Alas, the `dnf reposync` command will always create a directory called `azurelinux` (the name of the repo).